component/main.jsonnet

```
@@ -32,9 +34,10 @@ local patch_rbac = [
   '00_namespace': kube.Namespace(params.namespace) {
     metadata+: {
       labels+: {
-        [if ocp4 then 'openshift.io/cluster-monitoring']: 'true',
+        [if ocp4 && params.monitoring_enabled then 'openshift.io/cluster-monitoring']: 'true',
```

We need to add the cluster-monitoring label unconditionally, as otherwise with `monitoring.enabled=false` the ServiceMonitor (which is managed by the Helm chart) will be picked up by the user workload monitoring.

```suggestion
        [if ocp4 then 'openshift.io/cluster-monitoring']: 'true',
```

Alternatively, we could set `helm_values.enableMonitoring: ${patch_operator:monitoring:enabled}` in `class/defaults.yml`.